### ISSUE TITLE
Region::touchesRegion is broken

### ISSUE THREAD

**Issue by davisp** [libspatialindex/libspatialindex#55]
See here:

https://github.com/libspatialindex/libspatialindex/blob/master/src/spatialindex/Region.cc#L303-L307

The logic breaks down to:

    (x1 >= y1 + z && x1 <= y1 - z) || (x2 >= y2 + z && x2 <= y2 - z)

First, z is greater (`its std::numeric_limits<double>::epsilon()`) so we don't have to worry about negations on the logical comparisons or that it might be zero.

Second, make the simplifying assumption that `y1 == y2 == 0`

Which reduces our logic to:

    (x1 >= z && x1 <= -z) || (x2 >= z && x2 <= -z)

The only possible way that this logic could possibly evaluate to zero is if `x1 == x2 == z == 0` which we know to be impossible because z is a positive constant.

So basically no matter what two region objects are given, the function touchesRegion returns true.

Thankfully though the only cases where its used (at least in RTree code) returning true doesn't appear that it would break anything. It just doesn't allow for some optimizations in avoiding unnecessary MBR recalculations.

I would personally fix this by making the affirmative case return true:

    for (uint32_t i = 0; i < m_dimension; ++i)
    {
        if((x1 >= y1 - z && x1 <= y1 + z) || (x2 >= y2 - z && x2 <= y2 - z)) {
            return true;
        }
    }
    return false;

**Comment by davisp** [comment]
So there's more to this. Due to float rounding it is actually possible to return false from this function when (value + epsilon) and (value - epsilon) returns the same for both. But its still broken.
